doc/hardware/peripherals/eeprom/shell.rst

```diff
@@ -0,0 +1,91 @@
+.. _eeprom_shell:
+
+EEPROM Shell
+############
+
+.. contents::
+    :local:
+    :depth: 1
+
+Overview
+********
+
+The EEPROM shell provides an ``eeprom`` command with a set of subcommands for the :ref:`shell
+<shell_api>` module. It allows testing and exploring the :ref:`EEPROM <eeprom_api>` driver API
+through an interactive interface without having to write a dedicated application. The EEPROM shell
+can also be enabled in existing applications to aid in interactive debugging of EEPROM issues.
+
+In order to enable the EEPROM shell, the following :ref:`Kconfig <kconfig>` options must be enabled:
+
+* :kconfig:option:`CONFIG_SHELL`
+* :kconfig:option:`CONFIG_EEPROM`
+* :kconfig:option:`CONFIG_EEPROM_SHELL`
+
+For example, building the :ref:`hello_world` sample for the :ref:`native_sim` with the EEPROM shell:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: native_sim
+   :gen-args: -DCONFIG_SHELL=y -DCONFIG_EEPROM=y -DCONFIG_EEPROM_SHELL=y
+   :goals: build
+
+See the :ref:`shell <shell_api>` documentation for general instructions on how to connect and
+interact with the shell. The EEPROM shell comes with built-in help (unless
+:kconfig:option:`CONFIG_SHELL_HELP` is disabled). The built-in help messages can be printed by
+passing ``-h`` or ``--help`` to the ``eeprom`` command or any of its subcommands. All subcommands
+also support tab-completion of their arguments.
+
+.. tip::
+   All of the EEPROM shell subcommands take the name of a EEPROM peripheral as their first argument,
```

Comment:
nit

```suggestion
   All of the EEPROM shell subcommands take the name of an EEPROM peripheral as their first argument,
```

Reply:
Thank you for reviewing. I've fixed the spelling error and added a redirect as suggested.